Automattic/simplenote-android
# Update Tag List

### Fix
Update the tag list architecture by moving the code from `TagsListFragment` to `TagsActivity`.  This is part technical debt cleanup and part bug fix.  Prior to these changes, `TagsActivity` was merely a shell for `TagsListFragment` and consolidating those two classes would eliminate some minor complexity.  When the **_System default_** option is selected for the **_Theme_** setting on Android 10.0, enabling/disabling **_Dark theme_** at the operating system level would cause the **_Edit Tags_** screen background to follow the `Theme.Simplestyle.Splash` style rather than the `Theme.Simplestyle` the as expected.  See the animations below for illustration.

Before|After
-|-
<a href="https://user-images.githubusercontent.com/3827611/87373468-f65e9e00-c546-11ea-8244-840a8da1a49a.gif"><img src="https://user-images.githubusercontent.com/3827611/87373468-f65e9e00-c546-11ea-8244-840a8da1a49a.gif" width="300"></a>|<a href="https://user-images.githubusercontent.com/3827611/87373470-f78fcb00-c546-11ea-8a08-e0b0b20215a2.gif"><img src="https://user-images.githubusercontent.com/3827611/87373470-f78fcb00-c546-11ea-8a08-e0b0b20215a2.gif" width="300"></a>

### Test
1. Open navigation drawer.
2. Tap ***Edit*** button in **_Tags_** header.
3. Notice tags are listed with associated notes count.
4. Tap any tag in list.
5. Notice **_Rename Tag_** dialog is shown.
6. Tap ***Cancel*** button in **_Rename Tag_** dialog.
7. Tap ***Delete Tag*** button for tag with no associated note.
8. Notice tag is deleted from list.
9. Tap ***Delete Tag*** button for tag with associated note.
10. Notice **_Delete Tag_** dialog is shown.
11. Tap ***No*** button in **_Delete Tag_** dialog.

### Review
Only one developer is required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.
